node/src/reactor/validator.rs

```diff
@@ -416,6 +403,20 @@ impl reactor::Reactor for Reactor {
             .genesis
             .initial_block_height
             .unwrap_or(0);
+        let next_finalized_block = cmp::max(
+            latest_block
+                .as_ref()
+                .map(|block| block.height() + 1)
+                .unwrap_or(0),
+            initial_block_height,
+        );
```

Comment:
Could we just say `unwrap_or(initial_block_height)` or can it happen that `block` is earlier than that?

Reply:
I think it could happen. We could have some blocks in storage from before the restart, which would give us some last block in the linear chain, but still earlier than the initial block height.